# Description of difficult genes {#description}

```{r}
datasets1 <- c("GSE22260", "GSE41364", "GSE42146", "GSE50760", "GSE87340")
names(datasets1) <- paste0("dataset", c(4, 8, 7, 5, 6))
```

## Characteristics of DGs

Loading annotation for exons.

```{r annotation}
gtf.df <- readRDS("data/hg38geneModel.rds")
```


```{r tab1-mapers-separately, eval = F}
dg.lists <- readRDS("results/lists_dg_mappers_separately.rds")
ndg.lists <- readRDS("results/lists_ndg_mappers_separately.rds")
dg.lists <- transpose(map(dg.lists, transpose))$edgeR
ndg.lists <- transpose(map(ndg.lists, transpose))$edgeR

dg <- dg.lists %>% 
  enframe(name = "datasets") %>% 
  mutate(value = map(value, enframe, name = "mapper", value = "gene")) %>%
  unnest(value) %>%
  unnest(gene)

ndg <- ndg.lists %>% 
  enframe(name = "datasets") %>% 
  mutate(value = map(value, enframe, name = "mapper", value = "gene")) %>%
  unnest(value) %>%
  unnest(gene) 

dg$dg <- "yes"
ndg$dg <- "no"

df.tab1 <- bind_rows(dg, ndg) %>% 
  left_join(gtf.df, by = c("gene" = "gene"), relationship = "many-to-many") %>% 
  group_by(datasets, mapper, gene, dg) %>% 
  distinct(exon, .keep_all = TRUE) %>% 
  summarize(Ex.min = min(width), Ex.max = max(width), Ex.mean = mean(width), Ex.no = n(), Tr.length = sum(width)) %>% 
  group_by(datasets, mapper, dg) %>% 
  summarize(Ex.min = mean(Ex.min), Ex.max = mean(Ex.max), Ex.mean = mean(Ex.mean), Ex.no = mean(Ex.no), Tr.length = mean(Tr.length))
df.tab1
```


```{r old-tab1, eval=F}

dg.lists <- list()
ndg.lists <- list()
for (dataset in datasets)
{
  load(file = paste0("results/difficult_genes_", dataset, ".RData"))
  dg.lists[[dataset]] <- rownames(res.dane$edgeR$stats[which(res.dane$edgeR$stats$status.group == "Sig" & res.dane$edgeR$stats$status.mappers == "Sig"),])
  ndg.lists[[dataset]] <- rownames(res.dane$edgeR$stats[which(res.dane$edgeR$stats$status.group == "Sig" & res.dane$edgeR$stats$status.mappers == "nonSig"),])
}

dg <- dg.lists %>% 
  enframe(name = "datasets", value = "gene") %>% 
  unnest(gene)

ndg <- ndg.lists %>% 
  enframe(name = "datasets", value = "gene") %>% 
  unnest(gene)

dg$dg <- "yes"
ndg$dg <- "no"

df.tab1 <- bind_rows(dg, ndg) %>% 
  left_join(gtf.df, by = c("gene" = "gene"), relationship = "many-to-many") 

tab1 <- df.tab1 %>% 
  group_by(datasets, gene, transcript, dg) %>% 
  summarize(Ex.min = min(width), Ex.max = max(width), Ex.mean = mean(width), Ex.no = n(), Tr.length = sum(width)) %>% 
  ungroup() %>% 
  group_by(datasets, dg) %>% 
  summarize(across(where(is.numeric), ~ mean(.x, na.rm = TRUE)))

tab1
```


```{r current-tab1}
dg.lists <- list()
ndg.lists <- list()
for (dataset in datasets)
{
  load(file = paste0("results/difficult_genes_", dataset, ".RData"))
  dg.lists[[dataset]] <- rownames(res.dane$edgeR$stats[which(res.dane$edgeR$stats$status.group == "Sig" & res.dane$edgeR$stats$status.mappers == "Sig"),])
  ndg.lists[[dataset]] <- rownames(res.dane$edgeR$stats[which(res.dane$edgeR$stats$status.group == "Sig" & res.dane$edgeR$stats$status.mappers == "nonSig"),])
}

dg <- dg.lists %>% 
  enframe(name = "datasets", value = "gene") %>% 
  unnest(gene)

ndg <- ndg.lists %>% 
  enframe(name = "datasets", value = "gene") %>% 
  unnest(gene)

dg$Type <- "DG"
ndg$Type <- "non DG"

df.tab1 <- bind_rows(dg, ndg) %>% 
  left_join(gtf.df, by = c("gene" = "gene"), relationship = "many-to-many") 

tab1 <- df.tab1 %>% 
  group_by(datasets, gene, transcript, Type) %>% 
  summarize(Ex.min = min(width), Ex.max = max(width), Ex.median = median(width), Tr.length = sum(width), Ex.no = n(), feature1 = any(str_detect(feature, "pseudogene")), feature2 = any(str_detect(feature, "protein_coding")) ) %>% 
  ungroup() %>% 
  group_by(datasets, gene, Type) %>% 
  summarize(across(where(is.numeric), median), Tr.no = n(), feature1 = any(feature1), feature2 = any(feature2)) %>%
  ungroup() %>% 
  group_by(datasets, Type) %>% 
  summarize(across(where(is.numeric), median), Pseudogenes.percent = sum(feature1)/n()*100, Coding.percent = sum(feature2)/n()*100)
 
tab1$datasets[which(tab1$datasets == "dataset4")] <- "GSE22260"
tab1$datasets[which(tab1$datasets == "dataset5")] <- "GSE50760"
tab1$datasets[which(tab1$datasets == "dataset6")] <- "GSE87340"
tab1$datasets[which(tab1$datasets == "dataset7")] <- "GSE42146"
tab1$datasets[which(tab1$datasets == "dataset8")] <- "GSE41364"
```


```{r tab-characteristics}
header <- list()

header$pos <- list(0)
header$command <- c(
                    "\\multirow{2}{*}{GSE22260} & \\multirow{2}{*}{GSE41364} & \\multirow{2}{*}{GSE42146} & \\multirow{2}{*}{GSE50760} &\\multirow{2}{*}}{GSE87340} \\\\\n")

tab1 %>% 
  arrange(datasets) %>% 
  rename(Dataset = datasets) %>% 
  xtable(digits = 0,
         label = 'tab-char',
         caption = 'Characteristics of DG genes found in each dataset. Columns represent Ex.min - the length of the shortest exon, Ex.max - the length of the longest exon, Ex.mean - the mean exon length, Ex.no - the total number of exons as well as Tr.length - the transcript length. Each value was calculated as a mean value across all transcripts in datasets with division to DG and other genes.',
         align = "cc|c|cccccccc") %>%
  print(include.colnames = T,
        include.rownames = F,
        caption.placement = 'top',
        comment = FALSE,
        hline.after = c(-1,0,2,4,6,8,10),
        #add.to.row = header,
        NA.string = '--',
        sanitize.text.function = function(x){x},
        table.placement = 'ht!')
tab1
```

## Number of DGs

```{r dg-counts-all1, results="asis"}
for (dataset in datasets)
{
  dg <- readRDS(file = paste0("results/difficult_genes_", dataset, ".rds"))
  con.m <- matrix(,2,2*length(method))
  rownames(con.m) <- c('g.T', 'g.N')
  colnames(con.m) <- rep(c('m.T', 'm.N'),length(method))

  for (met in method)
  {
    con.m[1, (which(method == met)-1)*2+1] <- length(dg[[met]]$stats[which(dg[[met]]$stats$status.group == "Sig" & dg[[met]]$stats$status.mappers == "Sig"),1])/length(dg[[met]]$stats$status.group)
    con.m[1, (which(method == met)-1)*2+2] <- length(dg[[met]]$stats[which(dg[[met]]$stats$status.group == "Sig" & dg[[met]]$stats$status.mappers == "nonSig"),1])/length(dg[[met]]$stats$status.group)
    con.m[2, (which(method == met)-1)*2+1] <- length(dg[[met]]$stats[which(dg[[met]]$stats$status.group == "nonSig" & dg[[met]]$stats$status.mappers == "Sig"),1])/length(dg[[met]]$stats$status.group)
    con.m[2, (which(method == met)-1)*2+2] <- length(dg[[met]]$stats[which(dg[[met]]$stats$status.group == "nonSig" & dg[[met]]$stats$status.mappers == "nonSig"),1])/length(dg[[met]]$stats$status.group)
    
  }  
  cat("contingency table for methods edgeR, DESeq, limma + voom i limma + vst for dataset ", dataset, "\n")
  
  print(kable(con.m*100, digits = 2))
  cat("\n")
}

```

```{r dg-counts, results="asis"}

met <- "edgeR"

datasets <- datasets[match(datasets, names(datasets1))]
con.m <- matrix(,2,2*length(datasets))
rownames(con.m) <- c('yes', 'no')
colnames(con.m) <- rep(c('m.T', 'm.N'),length(datasets))

for (dataset in datasets)
{
  dg <- readRDS(file = paste0("results/difficult_genes_", dataset, ".rds"))
  
    con.m[1, (which(datasets == dataset)-1)*2+1] <- length(dg[[met]]$stats[which(dg[[met]]$stats$status.group == "Sig" & dg[[met]]$stats$status.mappers == "Sig"),1])/length(dg[[met]]$stats$status.group)
    con.m[1, (which(datasets == dataset)-1)*2+2] <- length(dg[[met]]$stats[which(dg[[met]]$stats$status.group == "Sig" & dg[[met]]$stats$status.mappers == "nonSig"),1])/length(dg[[met]]$stats$status.group)
    con.m[2, (which(datasets == dataset)-1)*2+1] <- length(dg[[met]]$stats[which(dg[[met]]$stats$status.group == "nonSig" & dg[[met]]$stats$status.mappers == "Sig"),1])/length(dg[[met]]$stats$status.group)
    con.m[2, (which(datasets == dataset)-1)*2+2] <- length(dg[[met]]$stats[which(dg[[met]]$stats$status.group == "nonSig" & dg[[met]]$stats$status.mappers == "nonSig"),1])/length(dg[[met]]$stats$status.group)
}

header <- list()

header$pos <- list(0,0,0)
header$command <- c("\\hline & \\multicolumn{10}{c}{Dataset}\\\\\n",
                    "\\cline{2-11} & \\multicolumn{2}{c|}{GSE22260} & \\multicolumn{2}{c|}{GSE41364} & \\multicolumn{2}{c|}{GSE42146} & \\multicolumn{2}{c|}{GSE50760} &\\multicolumn{2}{c}{GSE87340} \\\\\n", 
                    "\\hline \\diagbox{Groups}{Mappers} & yes & no & yes & no &  yes & no &  yes & no &  yes & no \\\\\n")
con.m <- con.m*100 
con.m %>%
  xtable(digits = 2,
         label = 'tab-per-difficult',
         caption = 'Percentage of significant genes due to mappers and groups across each dataset',
         align = "c|cc|cc|cc|cc|cc") %>%
  print(include.colnames = FALSE,
        include.rownames = TRUE,
        caption.placement = 'top',
        comment = FALSE,
        hline.after = c(0,2),
        add.to.row = header,
        NA.string = '--',
        sanitize.text.function = function(x){x},
        table.placement = 'ht!')

```

## Number of DGs with mappers separately

```{r dg-counts-mappers-separately,  results="asis"}
met <- "edgeR"
con.m <- matrix(,length(datasets),2*length(maper))
rownames(con.m) <- datasets1
colnames(con.m) <- rep(c('m.T', 'm.N'),length(maper))
dg.id <- list()

source("codes/differential_analysis_functions.R")
for (dataset in datasets)
{
  groups <- unlist(read.table(file = paste0('data/', dataset, '.csv')))
  dg <- readRDS(file = paste0("results/difficult_genes_", dataset, ".rds"))
  
  dg.id[[dataset]][[met]] <-
      dg[[met]]$stats %>%
      rownames_to_column("ID") %>%
      filter(status.mappers == "Sig" & status.group == "Sig") %>%
      arrange(pval.map, pval.gr)
  
  for (map in maper)
  {
    cc <- readRDS(file = paste0("data/", dataset, "_", map, ".rds"))
    
    res.data1 <- diffAnaAll1(cc, groups, method = met)
    res.data1 <- res.data1 %>% arrange(padj)
    deg.id <- res.data1$ID[which(res.data1$padj < 0.05)]
    
    con.m[which(datasets == dataset), (which(maper == map)-1)*2+1]  <- length(intersect(dg.id[[dataset]][[met]]$ID, deg.id))/length(res.data1$ID)*100
    
    con.m[which(datasets == dataset), (which(maper == map)-1)*2+2] <- length(intersect(deg.id, dg.id[[dataset]][[met]]$ID))/length(deg.id)*100
  }  
}
```


```{r tab2,  results="asis"}
header <- list()

header$pos <- list(0,0,0)
header$command <- c("\\hline \\multirow{3}{*}{Datasets} & \\multicolumn{6}{c}{Mappers}\\\\\n",
                    "\\cline{2-7} & \\multicolumn{2}{c|}{Hisat} & \\multicolumn{2}{c|}{Star} & \\multicolumn{2}{c|}{Subread} \\\\\n", 
                    "\\cline{2-7}  & \\% of all & \\% of DEG &  \\% of all & \\% of DEG &  \\% of all & \\% of DEG \\\\\n")

con.m %>%
  xtable(digits = 2,
         label = 'tab-per-difficult',
         caption = 'Percentage of significant genes due to mappers and groups across each dataset',
         align = "c|cc|cc|cc|") %>%
  print(include.colnames = FALSE,
        include.rownames = TRUE,
        caption.placement = 'top',
        comment = FALSE,
        hline.after = c(0,5),
        add.to.row = header,
        NA.string = '--',
        sanitize.text.function = function(x){x},
        table.placement = 'ht!')
```

## Venn diagrams 

```{r venns, results="asis"}
library(VennDiagram)

venn.colors <- brewer.pal(length(datasets), "Set1")
met <- "edgeR"

dg.list <- list()
for (dataset in datasets)
{
  dg <- readRDS(file = paste0("results/difficult_genes_", dataset, ".rds"))
  
    dg.list[[met]][[dataset]] <- rownames(dg[[met]]$stats)[which(dg[[met]]$stats$status.group == "Sig" & dg[[met]]$stats$status.mappers == "Sig")]
}

ID <- list()
b <- get.venn.partitions(x = list(
    A=dg.list[[met]][[1]],
    B=dg.list[[met]][[2]],
    C=dg.list[[met]][[3]],
    D=dg.list[[met]][[4]],
    E=dg.list[[met]][[5]]
  ))
a <- b$..count..
venn.plot <- draw.quintuple.venn(
    area1 = sum(a[c(1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31)]), 
    area2 = sum(a[c(1,2,5,6,9,10,13,14,17,18,21,22,25,26,29,30)]), 
    area3 = sum(a[c(1:4,9:12,17:20,25:28)]), area4 = sum(a[c(1:8,17:24)]), area5 = sum(a[1:16]),
    n12 = sum(a[c(1,5,9,13,17,21,25,29)]), n13 = sum(a[c(1,3,9,11,17,19,25,27)]), 
    n14 = sum(a[c(1,3,5,7,17,19,21,23)]), n15 = sum(a[c(1,3,5,7,9,11,13,15)]),
    n23 = sum(a[c(1,2,9,10,17,18,25,26)]), n24 = sum(a[c(1,2,5,6,17,18,21,22)]), 
    n25 = sum(a[c(1,2,5,6,9,10,13,14)]), n34 = sum(a[c(1:4,17:20)]), 
    n35 = sum(a[c(1:4,9:12)]), n45 = sum(a[c(1:8)]), 
    n123 = sum(a[c(1,9,17,25)]), n124 = sum(a[c(1,5,17,21)]), n125 = sum(a[c(1,5,9,13)]), 
    n134 = sum(a[c(1,3,17,19)]), n135 = sum(a[c(1,3,9,11)]), n145 = sum(a[c(1,3,5,7)]),
    n234 = sum(a[c(1,2,17,18)]), n235 = sum(a[c(1,2,9,10)]), 
    n245 = sum(a[c(1,2,5,6)]), n345 = sum(a[c(1:4)]),
    n1234 = sum(a[c(1,17)]), n1235 = sum(a[c(1,9)]),  n1245 = sum(a[c(1,5)]), 
    n1345 = sum(a[c(1,3)]), n2345 = sum(a[c(1:2)]),
    n12345 = sum(a[c(1)]),
    category = datasets1,
    fill = venn.colors, alpha = 0.30, cex = 1.2, fontfamily = "serif",
    rotation.degree = 360, margin = 0.1, main = "# of joint difficult genes between datasets",
    main.pos = c(0.5,0.8), height = 500,width = 500, resolution = 600, lwd = 0.5
)
grid.draw(venn.plot)
grid.newpage()
ID[[met]] <- b$..values..$`1`
```