--- 
title: "Difficult genes and their impact on RNA-Seq data analysis"
author:
- Alicja Szabelska-Beręsewicz
- Joanna Zyprych-Walczak
- Idzi Siatkowski
- Michał Okoniewski
output: 
      bookdown::pdf_book:
        includes:
          in_header: mystyles.sty
bibliography:
- book.bib
- packages.bib
documentclass: article
fontsize: 11pt
github-repo: Bioinformatics-Group-UP-Poznan/ambiguous-genes 
link-citations: yes
biblio-style: "apalike"
---


```{r packages, include=FALSE}
library(knitr)
library(xtable)
library(Cairo)
library(reshape)
library(tidyverse)
library(purrr)
library(RColorBrewer)
```

```{r setup1, include=FALSE}
options(xtable.comment = FALSE)
opts_chunk$set(echo = FALSE, message = FALSE, warning = FALSE,  tidy.opts = list(width.cutoff=60), tidy=TRUE, fig.width = 6, fig.height = 4.13, dpi = 300)
```

## Introduction

The issue being analysed is pinpointing the genes that cause systematically artifactual results in the analysis of RNA-seq. Such genes cannot be reliably measured and detected as differentially expressed. In particular the problem occurs, when popular genome aligners do not agree in the number and distribution of reads assigned to such genes. It causes confusion in reproducible data analysis. When such difficult genes are those of particular biological interest, it may distort the biological interpretation of the whole experiment. When difficult genes are the key ones in human metabolic pathways, the distorted results may be confusing for the further research in genomic personalised medicine.


```{r include=FALSE}
knitr::write_bib(c(
  .packages(), 'bookdown', 'knitr', 'rmarkdown'
), file = 'packages.bib')
```

