% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[
  11pt,
]{article}
\usepackage{lmodern}
\usepackage{amsmath}
\usepackage{ifxetex,ifluatex}
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
  \usepackage{amssymb}
\else % if luatex or xetex
  \usepackage{unicode-math}
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\IfFileExists{bookmark.sty}{\usepackage{bookmark}}{\usepackage{hyperref}}
\hypersetup{
  pdftitle={Difficult genes and their impact on RNA-Seq data analysis},
  pdfauthor={Alicja Szabelska-Beręsewicz; Joanna Zyprych-Walczak; Idzi Siatkowski; Michał Okoniewski},
  hidelinks,
  pdfcreator={LaTeX via pandoc}}
\urlstyle{same} % disable monospaced font for URLs
\usepackage{longtable,booktabs}
\usepackage{calc} % for calculating minipage widths
% Correct order of tables after \paragraph or \subparagraph
\usepackage{etoolbox}
\makeatletter
\patchcmd\longtable{\par}{\if@noskipsec\mbox{}\fi\par}{}{}
\makeatother
% Allow footnotes in longtable head/foot
\IfFileExists{footnotehyper.sty}{\usepackage{footnotehyper}}{\usepackage{footnote}}
\makesavenoteenv{longtable}
\usepackage{graphicx}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
% Set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother
\setlength{\emergencystretch}{3em} % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{5}
\usepackage{tabularx}
\usepackage{multirow}
\usepackage{diagbox}
\ifluatex
  \usepackage{selnolig}  % disable illegal ligatures
\fi
\usepackage[]{natbib}
\bibliographystyle{apalike}

\title{Difficult genes and their impact on RNA-Seq data analysis}
\author{Alicja Szabelska-Beręsewicz \and Joanna Zyprych-Walczak \and Idzi Siatkowski \and Michał Okoniewski}
\date{}

\begin{document}
\maketitle

{
\setcounter{tocdepth}{2}
\tableofcontents
}
\hypertarget{introduction}{%
\section{Introduction}\label{introduction}}

The issue being analysed is pinpointing the genes that cause systematically artifactual results in the analysis of RNA-seq. Such genes cannot be reliably measured and detected as differentially expressed. In particular the problem occurs, when popular genome aligners do not agree in the number and distribution of reads assigned to such genes. It causes confusion in reproducible data analysis. When such difficult genes are those of particular biological interest, it may distort the biological interpretation of the whole experiment. When difficult genes are the key ones in human metabolic pathways, the distorted results may be confusing for the further research in genomic personalised medicine.

\hypertarget{preliminary-analysis}{%
\section{Preliminary analysis}\label{preliminary-analysis}}

\hypertarget{heatmaps}{%
\subsection{Heatmaps}\label{heatmaps}}

\includegraphics{Summary-of-results_files/figure-latex/mapy-ciepla-1.pdf}

\hypertarget{barplots}{%
\subsection{Barplots}\label{barplots}}

\includegraphics{Summary-of-results_files/figure-latex/barploty-1.pdf}

\hypertarget{difficult_genes}{%
\section{Difficult genes}\label{difficult_genes}}

\includegraphics{Summary-of-results_files/figure-latex/wykresy-dziwnych-genow-1.pdf}

\hypertarget{description}{%
\section{Description of difficult genes}\label{description}}

contingency table for methods edgeR, DESeq, limma + voom i limma + vst for dataset dataset4

\begin{tabular}{l|r|r|r|r|r|r|r|r}
\hline
  & m.T & m.N & m.T & m.N & m.T & m.N & m.T & m.N\\
\hline
g.T & 10.09 & 28.28 & 2.32 & 12.47 & 9.24 & 28.80 & 7.29 & 24.1\\
\hline
g.N & 18.95 & 42.68 & 20.86 & 64.35 & 19.37 & 42.59 & 20.21 & 48.4\\
\hline
\end{tabular}

contingency table for methods edgeR, DESeq, limma + voom i limma + vst for dataset dataset5

\begin{tabular}{l|r|r|r|r|r|r|r|r}
\hline
  & m.T & m.N & m.T & m.N & m.T & m.N & m.T & m.N\\
\hline
g.T & 20.83 & 56.17 & 6.29 & 39.77 & 19.14 & 55.08 & 17.11 & 55.35\\
\hline
g.N & 10.26 & 12.73 & 15.30 & 38.62 & 11.71 & 14.07 & 12.62 & 14.92\\
\hline
\end{tabular}

contingency table for methods edgeR, DESeq, limma + voom i limma + vst for dataset dataset6

\begin{tabular}{l|r|r|r|r|r|r|r|r}
\hline
  & m.T & m.N & m.T & m.N & m.T & m.N & m.T & m.N\\
\hline
g.T & 21.08 & 50.96 & 10.15 & 42.51 & 20.60 & 49.19 & 19.12 & 49.40\\
\hline
g.N & 11.76 & 16.20 & 14.05 & 33.23 & 12.84 & 17.37 & 13.41 & 18.07\\
\hline
\end{tabular}

contingency table for methods edgeR, DESeq, limma + voom i limma + vst for dataset dataset7

\begin{tabular}{l|r|r|r|r|r|r|r|r}
\hline
  & m.T & m.N & m.T & m.N & m.T & m.N & m.T & m.N\\
\hline
g.T & 3.96 & 41.02 & 0.79 & 19.99 & 3.15 & 41.46 & 2.13 & 32.83\\
\hline
g.N & 8.97 & 46.05 & 6.74 & 72.47 & 8.46 & 46.93 & 9.17 & 55.86\\
\hline
\end{tabular}

contingency table for methods edgeR, DESeq, limma + voom i limma + vst for dataset dataset8

\begin{tabular}{l|r|r|r|r|r|r|r|r}
\hline
  & m.T & m.N & m.T & m.N & m.T & m.N & m.T & m.N\\
\hline
g.T & 16.01 & 62.67 & 9.32 & 56.37 & 15.45 & 62.35 & 14.41 & 63.72\\
\hline
g.N & 8.06 & 13.26 & 9.77 & 24.54 & 8.00 & 14.20 & 8.27 & 13.61\\
\hline
\end{tabular}

\begin{table}[ht!]
\centering
\caption{Percentage of significant genes due to mappers and groups across each dataset} 
\label{tab-per-difficult}
\begin{tabular}{c|cc|cc|cc|cc|cc}
   \hline & \multicolumn{10}{c}{Dataset}\\
 \cline{2-11} & \multicolumn{2}{c|}{GSE22260} & \multicolumn{2}{c|}{GSE41364} & \multicolumn{2}{c|}{GSE42146} & \multicolumn{2}{c|}{GSE50760} &\multicolumn{2}{c}{GSE87340} \\
 \hline \diagbox{Groups}{Mappers} & yes & no & yes & no &  yes & no &  yes & no &  yes & no \\
 \hline
yes & 10.09 & 28.28 & 16.01 & 62.67 & 3.96 & 41.02 & 20.83 & 56.17 & 21.08 & 50.96 \\ 
  no & 18.95 & 42.68 & 8.06 & 13.26 & 8.97 & 46.05 & 10.26 & 12.73 & 11.76 & 16.20 \\ 
   \hline
\end{tabular}
\end{table}

\begin{table}[ht!]
\centering
\caption{Percentage of significant genes due to mappers and groups across each dataset} 
\label{tab-per-difficult}
\begin{tabular}{c|cc|cc|cc|}
   \hline \multirow{3}{*}{Datasets} & \multicolumn{6}{c}{Mappers}\\
 \cline{2-7} & \multicolumn{2}{c|}{Hisat} & \multicolumn{2}{c|}{Star} & \multicolumn{2}{c|}{Subread} \\
 \cline{2-7}  & \% of all & \% of DEG &  \% of all & \% of DEG &  \% of all & \% of DEG \\
 \hline
GSE22260 & 0.10 & 17.93 & 0.25 & 27.11 & 0.20 & 22.75 \\ 
  GSE41364 & 11.60 & 23.38 & 13.34 & 26.19 & 12.62 & 25.39 \\ 
  GSE42146 & 11.73 & 20.70 & 11.61 & 20.43 & 14.52 & 25.92 \\ 
  GSE50760 & 10.08 & 18.04 & 9.90 & 17.67 & 11.40 & 20.63 \\ 
  GSE87340 & 0.44 & 4.81 & 0.44 & 4.22 & 0.77 & 7.27 \\ 
   \hline
\end{tabular}
\end{table}

\includegraphics{Summary-of-results_files/figure-latex/venny-1.pdf}

\hypertarget{machine-learning}{%
\section{Machine learning}\label{machine-learning}}

\hypertarget{classification-errors}{%
\subsection{Classification errors}\label{classification-errors}}

\includegraphics{Summary-of-results_files/figure-latex/wykres-polaczonego-klasyfikatora-1.pdf}

\begin{table}[ht!]
\centering
\caption{Average AUC values for 10 simulations for considered datasets and mappers} 
\label{tab-AUC}
\begin{tabular}{c|c|cc|cc|cc}
   \hline \multirow{3}{*}{Dataset} & \multirow{3}{*}{No of pred} & \multicolumn{6}{c}{Mapper/Difficultness} \\
 \cline{3-8}  & & \multicolumn{2}{c|}{Hisat}& \multicolumn{2}{c|}{Star} & \multicolumn{2}{c}{Subread}\\
  & & yes & no &  yes & no &  yes & no \\
 \hline
 & 10 & 0.575 & 0.893 & 0.569 & 0.904 & 0.717 & 0.945 \\ 
  GSE22260 & 15 & 0.598 & 0.901 & 0.636 & 0.884 & 0.668 & 0.939 \\ 
   & 20 & 0.610 & 0.903 & 0.905 & 0.879 & 0.908 & 0.913 \\ 
   \hline
 & 18 & 0.890 & 0.881 & 0.893 & 0.814 & 0.899 & 0.877 \\ 
  GSE41364 & 27 & 0.910 & 0.878 & 0.908 & 0.811 & 0.906 & 0.870 \\ 
   & 36 & 0.897 & 0.858 & 0.906 & 0.795 & 0.901 & 0.865 \\ 
   \hline
 & 14 & 1.000 & 0.991 & 1.000 & 1.000 & 1.000 & 0.998 \\ 
  GSE42146 & 22 & 1.000 & 1.000 & 1.000 & 1.000 & 1.000 & 1.000 \\ 
   & 29 & 1.000 & 1.000 & 1.000 & 1.000 & 1.000 & 1.000 \\ 
   \hline
 & 4 & 0.773 & 0.977 & 0.913 & 0.838 & 0.899 & 0.788 \\ 
  GSE50760 & 6 & 0.795 & 0.988 & 0.908 & 0.970 & 0.894 & 0.874 \\ 
   & 8 & 0.865 & 0.995 & 0.948 & 0.969 & 0.918 & 0.811 \\ 
   \hline
 & 3 & 1.000 & 0.886 & 1.000 & 0.930 & 1.000 & 0.958 \\ 
  GSE87340 & 4 & 1.000 & 0.944 & 1.000 & 0.998 & 1.000 & 1.000 \\ 
   & 6 & 1.000 & 0.945 & 1.000 & 0.990 & 1.000 & 0.996 \\ 
  \end{tabular}
\end{table}

  \bibliography{book.bib,packages.bib}

\end{document}
